java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/model/typesystem/ReferenceTypeUsageImpl.java

```diff
@@ -34,6 +35,12 @@ public ReferenceTypeUsageImpl(TypeDeclaration typeDeclaration, TypeSolver typeSo
 
     public ReferenceTypeUsageImpl(TypeDeclaration typeDeclaration, List<TypeUsage> typeParameters, TypeSolver typeSolver) {
         super(typeDeclaration, typeParameters, typeSolver);
+        if (!typeDeclaration.getTypeParameters().isEmpty() && typeParameters.isEmpty()) {
```

Comment:
This super.typeParameters surprises me. Maybe you could instead have a static helper method which process the typeParameters and return them. You could call in this method in the super call:

```
super(typeDeclaration, myTypeParametersProcessor(typeDeclaration, typeParameters), typeSolver)
```